### PR TITLE
dont use NO_DATA_VALUES in bilinear filter

### PR DESCRIPTION
**Pull request by kevinkreiser**
not ready to ship this yet as i want to finish my test that confirms the expected result near data boundaries. test forth coming.

### CODE REVIEW

**Comment by kevinkreiser** [comment]
this pr is ready to be reviewed now with a test that makes sure only those pixels with values are used when doing the bilinear filter otherwise we scale the weightings accordingly

File: src/skadi/sample.cc [hunk 1]
```diff
@@ -115,21 +115,42 @@ namespace skadi {
...
+      //LOG_INFO('{' + std::to_string((y + 1) * HGT_DIM + x) + ',' + std::to_string(c) + '}');
+      //LOG_INFO('{' + std::to_string((y + 1) * HGT_DIM + x + 1) + ',' + std::to_string(d) + '}');
+      value += (c * c_coef + d * d_coef);
     }
-    return value;
+    //if we are missing everything then give up
+    if(!has_data)
+      return NO_DATA_VALUE;
+    //if we were missing some we need to adjust by that
+    return value / (1 - adjust);
```
Comment (kevinkreiser) [inline src/skadi/sample.cc]: can simplify significantly if we just add the coefficients that counted instead of the ones that didnt. will do that in another commit tomorrow

**Comment by dnesbitt61** [comment]
:shipit:
